STITCH.md

```diff
@@ -0,0 +1,153 @@
+Important
+=========
+
+This document is based on suppositions about the Twitter's Stitch library, since it was announced[[1]](https://www.youtube.com/watch?v=VVpmMfT8aYw)[[2]](https://www.youtube.com/watch?v=bmIxIslimVY), but isn't open sourced yet.
+
+Introduction
+============
+
+Considering the absence of a timeline for the Stitch open sourcing, we decided to start the Clump project. The initial goal was to have a similar implementation, but the project evolved to provide an approach more adherent to some use-cases we have in mind. This document presents what we believe to be the main differences between Stitch and Clump.
+
+1. Optional fetches
+===================
+
+Stitch fetch sources are defined using objects:
+
+```scala
+case object FetchTracks extends SeqGroup[Id, Track] {
+	def run(ids: Seq[Id]): Future[Seq[Track]] = 
+		tracksService.fetch(ids)
+}
+```
+
+For each ```Id``` input, the ```SeqGroup.run``` must provide one ```Track``` instance. The signature doesn't represent well the cases where some ```Track``` instances may be missing from the ```tracksService```'s response. As an alternative, Stitch allows to define the response as optional:
+
+```scala
+case object FetchTracks extends SeqGroup[Id, Option[Track] {
+	def run(ids: Seq[Id]): Future[Seq[Option[Track]]] = 
+		tracksService.fetch(ids)
+}
+```
+
+Basically, we can say that the Stitch's API is optimized for non-optional fetches. In the other hand, Clump is optimized for optional fetches:
+
+```scala
+val tracksSource = Clump.source(tracksService.fetch)(_.trackId)
+```
+
+The source instance applies the transformations to represent each fetch result as an ```Option``` by extracting the track ids. 
+
+This means that the final clump value is also optional:
+
+```scala
+val clump = tracksSource.get(222)
+val result: Future[Option[Track]] = clump.get
+```
+
+3. Composition
+==============
+
+Given the optimization to deal with optional fetches, the ```Clump``` compositions is also different from Stitch. They have semantics similar to relational database's joins, where not found joined elements make the tuple be filtered-out.
```

Comment:
I never thought of it like a database join, that's a really awesome way to describe it